### PR TITLE
7000 mpconfig solr

### LINKED ISSUE #3212: Support multi-core Solr instances
Currently, Dataverse assumes the URL you're entering for Solr is a single-core instance (as evidenced by the code adding /solr to the config url)

https://github.com/IQSS/dataverse/blob/v4.4/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java#L91

If the config option was changed to be the full endpoint URL to Solr, (so localhost:8983/solr instead of just localhost:8983) then Dataverse could support multi-core instances which have URLs that look like localhost:8983/solr/corename/

Although this config change will break existing installs, if there was an upgrade detection process, then you could easy just add /solr to the end of the existing config value and everything would be functional once again.

### PR DESCRIPTION
**Pull request by poikilotherm**
**What this PR does / why we need it**:
For unknown reasons, the Solr endpoint has been made configurable via a Database Setting only. This is very cumbersome for testing and container usage. Providing simple defaults via MPCONFIG takes lots of this load away. It also properly decouples creating the connection to Solr from the Database connection.

This PR is fully backward compatible with the old behaviour - anyone already having the database setting will let that setting take precedence!

This PR also very much enhances the configuration of the Solr connection, as it allows for the first time to choose transport protocol, core name and/or path, which makes it much more compatible with Solr installations that differ from our classic installations. (It does, however, not yet add support for SolrCloud).

Relates to #7000
Closes #3212

**Which issue(s) this PR closes**:
None.

**Special notes for your reviewer**:
- [ ] Please give advice if the installer should be changed to not use the database setting anymore.
- [ ] Please give advice if a release note update should be included.

**Suggestions on how to test this**:
Play around with the settings described in the added docs and watch it work. Or take a look at the nice test examples, also using the new shiny `@JvmSetting` test extension!

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
Not really.

**Is there a release notes update needed for this change?**:
Maybe? Dunno. Please leave comment.

**Additional documentation**:
Included.

### CODE REVIEW

**Comment by mreekie** [comment]
added to sprint Dec 15, 2022

File: src/main/java/edu/harvard/iq/dataverse/search/SolrClientService.java [hunk 1]
```diff
@@ -30,6 +33,7 @@
 @Singleton
 public class SolrClientService {
     private static final Logger logger = Logger.getLogger(SolrClientService.class.getCanonicalName());
+    private static final Config config = ConfigProvider.getConfig();
```
[the remark this answers is not in the document]
Comment (pdurbin) [reply src/main/java/edu/harvard/iq/dataverse/search/SolrClientService.java:36]: @poikilotherm should this still be removed? If so, please go ahead.

File: src/main/resources/META-INF/microprofile-config.properties [hunk 2]
```diff
@@ -3,6 +3,15 @@
 dataverse.version=${project.version}
 dataverse.build=
 
+# SEARCH INDEX
+dataverse.solr.host=localhost
+# Activating mp config profile -Dmp.config.profile=ct changes default to "solr" as DNS name
+%ct.dataverse.solr.host=solr
```
Comment (pdurbin) [inline src/main/resources/META-INF/microprofile-config.properties:9]: What is this `%` for? Is it a typo? If not, can we please have a comment about out? Or is it documented in the guides already?

File: doc/sphinx-guides/source/installation/config.rst [hunk 3]
```diff
@@ -1783,6 +1838,21 @@ To facilitate large file upload and download, the Dataverse Software installer b
...
 and restart Payara to apply your change.
 
+mp.config.profile
++++++++++++++++++
+
+MicroProfile Config 2.0 defines the `concept of "profiles" <https://download.eclipse.org/microprofile/microprofile-config-2.0/microprofile-config-spec-2.0.html#configprofile>`_.
+They can be used to change configuration values by context. This is used in Dataverse to change some configuration
+defaults when used inside container context rather classic installations.
+
+As per the spec, you will need to set the configuration value ``mp.config.profile`` to ``ct`` as early as possible.
```
Comment (pdurbin) [inline doc/sphinx-guides/source/installation/config.rst:1848]: What does "ct" stand for?

**Review by pdurbin — COMMENTED** [review]
Just some quick comments.

File: doc/sphinx-guides/source/installation/config.rst (hunk 3 above)
Comment (poikilotherm) [reply doc/sphinx-guides/source/installation/config.rst:1848]: "ct" means "container". I've been using this in quite a few places now, e.g. the Maven profile to build the images etc.

File: src/main/resources/META-INF/microprofile-config.properties (hunk 2 above)
Comment (poikilotherm) [reply src/main/resources/META-INF/microprofile-config.properties:9]: Nope, this is not a typo, but activating using the value when profile "ct" is active (see your question/comment above)

https://download.eclipse.org/microprofile/microprofile-config-3.0/microprofile-config-spec-3.0.html#_how_config_profile_works

File: src/main/java/edu/harvard/iq/dataverse/search/SolrClientService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (poikilotherm) [reply src/main/java/edu/harvard/iq/dataverse/search/SolrClientService.java]: Done.

File: src/main/resources/META-INF/microprofile-config.properties (hunk 2 above)
Comment (pdurbin) [reply src/main/resources/META-INF/microprofile-config.properties:9]: Ok, makes more sense now. Thanks.

File: doc/sphinx-guides/source/installation/config.rst (hunk 3 above)
Comment (pdurbin) [reply doc/sphinx-guides/source/installation/config.rst:1848]: Ok, thanks.

**Review by pdurbin — APPROVED** [review]
I didn't test this but the API tests are passing and the code changes make sense. Approved.

**Comment by pdurbin** [comment]
@poikilotherm more merge conflicts, as expected since other MPCONFIG PRs have been merged. Can you please resolved them?

**Comment by kcondon** [comment]
Ready to merge except for the merge conflicts.

**Comment by poikilotherm** [comment]
All set! :smile:

**Pull request merged**